### ISSUE TITLE
How can I change keyframe's level?

### ISSUE THREAD

**Issue by hatwearingdumb** [opentoonz/opentoonz#4728]
### Ask a question related to OpenToonz

And I'm not talking about the "Replace level" options, I need to **transfer** all of my level's frames onto their own levels. This had became a big issue when I tried to change the color of a Toonz Raster Level, and then realised that my other columns are affected too, since I was originally doing the whole thing on fewer of them.

![image](https://user-images.githubusercontent.com/79970290/214882619-5809e6c7-cf4e-4462-917b-5586d6bf0523.png)

It would be really handy if there was an option, for automatic column-to-level separation. So, for example, when I move or copy a frame from Col1 to Col2, the keyframe changes from being on the A level, to being on the B level automatically. Or at least give me an option to transfer all selected frames to another level.

**Comment by DarrenTAnims** [comment]
When you talk about keyframes, is it right that you're not talking about keys from the animation tool? On first reading, I thought you was referring to animation tool keys.
It's not likely to have automatic moving of drawings from one level to another. If you want drawings to be in a different level then you should just copy them with the selection tool.

**Comment by RodneyBaker** [comment]
We can use `Cells > Clone Cells` in the menu to create new (independent) copies of levels.
If it's something done often we can set a shortcut key to that.  Alt + C works well as an assignment there.

![image](https://user-images.githubusercontent.com/2576432/214906089-f2112408-4c07-468f-911f-a09235600c40.png)

**Comment by HatWearingMann** [comment]
> We can use `Cells > Clone Cells` in the menu to create new (independent) copies of levels. If it's something done often we can set a shortcut key to that. Alt + C works well as an assignment there.
> 
> ![image](https://user-images.githubusercontent.com/2576432/214906089-f2112408-4c07-468f-911f-a09235600c40.png)
> 
> Added: We can also use Merge (Merge TLV for TLV levels) to transfer to a new level in a column. That new column will nee to have empty frames in most cases.

The newly cloned cells still aren't on the same level though.
![image](https://user-images.githubusercontent.com/79970290/215197417-d71d77bd-6416-4cbf-a4d2-075ffdcb3d3d.png)

Also Darren, I meant levels as if the .tlv files. I want to move existing drawings to them, and not to new collumns.

**Comment by RodneyBaker** [comment]
I'm still not fully tracking but we are getting there.
Keep in mind that multiple 'levels' can be inside the same column.
A 'column' simply being a container for levels.
I see in your image that you have a D level and a B level.

To 'merge' columns there are several means... 
We can actually merge them in a number of ways...using Merge TLV or Merge...  depending on the level type.
We can copy/paste.
We can collapse columns containing levels into a sub-xsheet column that contains all those columns and levels we have purposed to collect into that column.
We can render out various levels/columns and use that combined set of levels as a single level.

In short, two independent levels cannot be the same level unless we merge them in some fashion.

I suppose we could add a means to automate the copy/paste process a little more so that we could select any number of cells and Opentoonz would interatively move through those frames and copy/paste them all into either the first or last frame selected.

Unless columns are collapsed, different levels cannot occupy the same cell of a column (e.g. row in Timeline).

**Comment by RodneyBaker** [comment]
Short of something we are missing I'll guess your best way forward is to use Merge TLV levels.
This because you want to retain the ability to adjust colors in a way that only TLV (toonz raster) levels will allow.
Methods of copy/paste would work as well since we are dealing with TLV levels as both source and target.

**Comment by HatWearingMann** [comment]
> To 'merge' columns there are several means... We can actually merge them in a number of ways...using Merge TLV or Merge...

Both are grayed out when I select a column.

**Comment by HatWearingMann** [comment]
Okay I just figured out how to do what I wanted to:

![image](https://user-images.githubusercontent.com/79970290/215290599-16ae1bfb-24e4-4cc9-93a4-61462a591c46.png)
![image](https://user-images.githubusercontent.com/79970290/215290605-c2d6b55e-0b29-405d-ada0-39ea5c1e0539.png)
![image](https://user-images.githubusercontent.com/79970290/215290695-156e91b9-257a-4af5-a4cf-b9c83ba46631.png)

Basically, I want to do that, but automatically when I move a frame to another collumn (so every collumn has only one level assigned to it).

**Comment by RodneyBaker** [comment]
If grayed out that is simply an indicator that some criteria for merger is not present.
Also, we are merging levels contained in columns so the levels need to be in columns that you want to combine.

There are some specific things we want to do when merging... and different level types have slightly different criteria.

With toonz raster (TLV) we usually want to Save All first before merging because merging is a destructive process and we might want to keep the old levels that are being merged in case we want to return to them.

Importantly, we need to select columns that need to be merged.  Selecting cells will usually result in an error message suggesting selection of columns.  Selecting columns that contain different level types will result in the option being grayed out... but I don't think that is your issue because you are only using TLV levels.

**Comment by RodneyBaker** [comment]
As we can sense a feature request here we'll add a feature request label.
The more information on what success would look like the better.

**Comment by HatWearingMann** [comment]
Well, I don't know what else to add, except for showing off my desired result:
![image](https://user-images.githubusercontent.com/79970290/215351152-b3da9cd6-40be-4851-8f04-78e2a1d1a8b0.png)

**Comment by artisteacher** [comment]
Are you trying to clone the cells from multiple levels in a single column and have them automatically create a single cloned level in a new column?

**Comment by HatWearingMann** [comment]
I am trying to move the cells from multiple levels to a single column and have them moved automatically to a single level, that was previously assigned to that column.

**Comment by hatwearingdumb** [comment]
Hah, I just happened to reinstall the program today, I'll be sure to test it :)

**Comment by hatwearingdumb** [comment]
I think the issue back then was I expected each column to be assigned to a single level, so you wouldn't end up with A, B and C in a single column. It's still the case, since there are no downsides to it, besides the timeline looking a little messy.

What bothered me more, was how once you remove all the key frames of a level, from the timeline, its name is not assignable anymore, due to it still existing in the scene's directory.

![2024-07-13-28_OpenToonz](https://github.com/user-attachments/assets/89ab3be8-add8-4c11-ac57-7434d96a96b5)
![2024-07-13-29_explorer](https://github.com/user-attachments/assets/d6649830-5def-4b8b-88b3-5b31a0c792fb)

So not a real problem, but rather a wake to my inner pedantic.
